### PR TITLE
bpf: Fix sockops compile/verify on newer LLVM

### PR DESCRIPTION
**Pull request by joestringer**
Older versions of Clang/LLVM (7.0 and earlier) compile this line into a
32-bit read followed by a shift:

```
  r8 = *(u32 *)(r6 + 64)
  r8 >>= 16
```

Newer versions of Clang/LLVM will compile this instead into a single
16-bit read into an offset from the context like this:

```
  r8 = *(u16 *)(r6 + 66)
```

Unfortunately, the kernel verifier (at least as of v5.0, possibly newer)
does not currently support reads into 'struct bpf_sock_ops' at such
offsets, which leads to verifier errors when attempting to load the
program when compiled on newer compilers.

Fix it by forcing a 32-bit read and shifting it.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9336)

### CODE REVIEW

File: bpf/sockops/bpf_sockops.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ static __always_inline void sk_extract4_key(struct bpf_sock_ops *ops,
 	key->family = ENDPOINT_KEY_IPV4;
 
 	key->sport = (bpf_ntohl(ops->local_port) >> 16);
-	key->dport = ops->remote_port >> 16;
+	key->dport = READ_ONCE(ops->remote_port) >> 16;
```
Comment (jrfastab) [inline bpf/sockops/bpf_sockops.h:68]: add a comment here about clang 7.0 generating a u16 read that is unsupported on some kernels.

Comment (joestringer) [reply bpf/sockops/bpf_sockops.h:68]: 7.1 or 8.0, but yeah. I'll fix this up

**Review by jrfastab — APPROVED** [review]
LGTM just add a small note in the code as a hint to the reader why we use READ_ONCE.

**Comment by joestringer** [comment]
test-me-please

**Comment by ianvernon** [comment]
I'm removing the need to backport this to v1.5, we are only doing critical security fixes in v1.5 now.

**Comment by joestringer** [comment]
test-me-please

**Pull request merged**